ts/features/messages/utils/messages.ts

```diff
@@ -222,24 +219,25 @@ const extractCTA = (
     text,
     FM.test,
     O.fromPredicate(identity),
-    O.chainNullableK(() => {
-      try {
-        return FM<MessageCTA>(text).attributes;
-      } catch (e) {
-        trackMessageCTAFrontMatterDecodingError(serviceId);
-        return null;
-      }
-    }),
-    O.chain(attrs =>
+    O.chain(() =>
+      pipe(
+        E.tryCatch(() => FM<MessageCTA>(text).attributes, E.toError),
+        E.mapLeft(() => trackMessageCTAFrontMatterDecodingError(serviceId)),
+        O.fromEither
+      )
+    ),
+    O.chain(attributes =>
       pipe(
-        attrs[getRemoteLocale()],
+        attributes[getRemoteLocale()],
         CTAS.decode,
-        E.fold(
-          _ => O.none,
+        O.fromEither,
+        O.chain(ctas => {
```

Comment:
What about
```
O.filter(ctas => hasCtaValidActions(ctas, serviceMetadata))
```
instead of
```
O.chain(ctas => {
  // check if the decoded actions are valid
  if (hasCtaValidActions(ctas, serviceMetadata)) {
    return O.some(ctas);
  }
  return O.none;
})
```
?